src/test/java/hudson/plugins/xvnc/XvncTest.java

```diff
@@ -134,6 +147,118 @@ public void avoidNpeAfterDeserialiation() throws Exception {
+                FilePath[] list = build.getWorkspace().list(".Xauthority-*");
+                assertEquals("There should be one Xauthority file in the workspace", 1, list.length);
+                assertEquals(build.getWorkspace().getRemote(), list[0].getParent().getRemote());
+                return true;
+            }
+        });
+        j.buildAndAssertSuccess(job);
+
+        FilePath[] list = job.getLastBuild().getWorkspace().list(".Xauthority-*");
+        assertEquals("The Xauthority file should be removed", 0, list.length);
+    }
+
+    @Test
+    public void testXauthorityInSlaveFsRoot() throws Exception {
+        final DumbSlave slave = j.createOnlineSlave();
+
+        FreeStyleProject job = j.jenkins.createProject(FreeStyleProject.class, "job A");
+        job.setAssignedNode(slave);
+
+        runXvnc(job, true);
+
+        job.getBuildersList().add(new TestBuilder() {
+            @Override
+            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+                FilePath[] list = slave.getRootPath().list(".Xauthority-*");
+                assertEquals("There should be one Xauthority file in the slave fs root", 1, list.length);
+                assertEquals(slave.getRootPath().getRemote(), list[0].getParent().getRemote());
+
+                list = build.getWorkspace().list(".Xauthority-*");
+                assertEquals("There should be no Xauthority file in the workspace", 0, list.length);
+
+                return true;
+            }
+        });
+        j.buildAndAssertSuccess(job);
+
+        FilePath[] list = slave.getRootPath().list(".Xauthority-*");
+        assertEquals("The Xauthority file should be removed", 0, list.length);
+    }
+
+    @Test
+    public void testXauthorityInTemp() throws Exception {
+        final File base = new File(System.getProperty("java.io.tmpdir"), "some bad path");
+        base.mkdirs();
+        final TemporaryDirectoryAllocator directoryAllocator = new TemporaryDirectoryAllocator(base);
+        try {
+            final DumbSlave slave = createTweakedSlave(directoryAllocator);
+            final FilePath tmpPath = new FilePath(slave.getChannel(), System.getProperty("java.io.tmpdir"));
+            FreeStyleProject job = j.jenkins.createProject(FreeStyleProject.class, "job A");
+            job.setAssignedNode(slave);
+
+            runXvnc(job, true);
+
+            job.getBuildersList().add(new TestBuilder() {
+                @Override
+                public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+
+                    FilePath[] list = tmpPath.list(".Xauthority-*");
+                    assertEquals("There should be one Xauthority file in tmp", 1, list.length);
```

Comment:
Then a warning is written to the build log but tmp is still chosen because we are out of options, the user might have a different vncserver command for example so it could still work.